src/Plutarch/Extra/ScriptContext.hs

```diff
@@ -439,3 +451,60 @@ pfindOwnInput = phoistAcyclic $
     matches = phoistAcyclic $
       plam $ \outref txininfo ->
         outref #== pfield @"outRef" # txininfo
+
+{- | Safely convert a 'ValidatorHash' into a 'TokenName'. This can be useful for tagging
+     tokens for extra safety.
+
+     @since 3.14.1
+-}
+validatorHashToTokenName :: ValidatorHash -> TokenName
+validatorHashToTokenName = coerce
+
+{- | Safely convert a 'PValidatorHash' into a 'PTokenName'. This can be useful for tagging
+     tokens for extra safety.
+
+     @since 3.14.1
+-}
+pvalidatorHashToTokenName ::
+  forall (s :: S).
+  Term s PValidatorHash ->
+  Term s PTokenName
+pvalidatorHashToTokenName = punsafeCoerce
+
+{- | Safely convert a 'PScriptHash' into a 'PTokenName'. This can be useful for tagging
+     tokens for extra safety.
+
+     @since 3.14.1
+-}
+scriptHashToTokenName :: ScriptHash -> TokenName
+scriptHashToTokenName = coerce
+
+{- | Safely convert a 'PScriptHash' into a 'PTokenName'. This can be useful for tagging
+     tokens for extra safety.
+
+     @since 3.14.1
+-}
+pscriptHashToTokenName ::
+  forall (s :: S).
+  Term s PScriptHash ->
+  Term s PTokenName
+pscriptHashToTokenName = punsafeCoerce
+
+{- | Find a redeemer in the redeemer map and convert it.
+
+    @since 3.14.1
+-}
+ptryFromRedeemer ::
+  forall (r :: PType) (s :: S).
+  (PTryFrom PData r) =>
+  Term
+    s
+    ( PScriptPurpose
+        :--> PMap 'Unsorted PScriptPurpose PRedeemer
```

Comment:
It can accept Unsorted and Sorted.

Reply:
The intended use case is to pass in the `redeemers` field of `PTxInfo`, which is defined as:
```hs
"redeemers" := PMap 'Unsorted PScriptPurpose PRedeemer
```
I should mention it in the doc.